File: Project_02_C#/project_2_CSharp/project_2_CSharp/App/YoutubeApp.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace project_2_CSharp.App
{
    class YoutubeApp: IAppleCompatible
    {
        private string appName = "Youtube for IOS";
        private string version = "15.2";

        public string downloading()
        {
            return "Downloading " + ToString();
        }
        public string OSCompatible
        {
            get
            {
                return "IOS";
            }
        }

        public override string ToString()
        {
            return "app name: " + appName + ", Version: " + version;
        }

        public bool check(string os)
        {
            if (os == this.OSCompatible)
            {
                return true;
            }
            return false;
        }
    }
}

File: project_2_java/src/app/YoutubeApp.java
package app;

public class YoutubeApp implements IAppleCompatible
{
	private String appName = "Youtube for IOS";
	private String version = "15.2";

	public final String downloading()
	{
		return "Downloading " + toString();
	}
	public final String getOSCompatible()
	{
		return "IOS";
	}

	@Override
	public String toString()
	{
		return "app name: " + appName + ", Version: " + version;
	}

	public final boolean check(String os)
	{
		if (this.getOSCompatible().equals(os))
		{
			return true;
		}
		return false;
	}
}
